### ISSUE TITLE
Built-in SMTP server / MTA

### ISSUE THREAD

**Issue by knadh** [knadh/listmonk#528]
If listmonk comes with a built-in SMTP server, it may be a big value add for the long tail of users with small to medium sized mailing lists. Of course, there may be SPF/DKIM/DMARC/IP reputation hassles to deal with, but it would make listmonk truly standalone and self-contained. [emersion/go-smtp](https://github.com/emersion/go-smtp) looks like a good lib for this.

I'm still ruminating on the idea. Opening this issue in case anyone has any thoughts.

**Comment by knadh** [comment]
>  IP/Domain reputation is a big deal. So are DKIM/SPF/DMARC What is the point of sending if the email doesn't reach the inbox?

That's right, but if somebody wants to put in the effort, they can (we do this at work successfully).

> it is foolhardy to send from a VPS/host and try to manage their own IP reputation.

That could be said of self-hosting a mailing list manager too!

**Comment by ChrisTG742** [comment]
> > IP/Domain reputation is a big deal. So are DKIM/SPF/DMARC What is the point of sending if the email doesn't reach the inbox?

I second this. As phishing attempts raise constantly, the automated technics to separate ham from spam are also getting more and more picky. We had to do a lot of efforts to finally get the automated mails get into the users's inboxes instead of being declared as spam every time. In the end we decided to put all mail traffic trough a dedicated mail-gateway which has a public IP that is trusted by other mailservers. That's the IP-blacklist issue one simply can't fight against locally. Many providers consider SMTP-hosts running on dial-ins or customer IPs to be untrustworthy. So I can imagine how much un-solvable support regarding the SMTP-service solely this circumstance could generate.

TL;DR: I'm consitent with @andreafa and also wouldn't bother to tilt at windmills in this case.

**Comment by bamboowonder** [comment]
@knadh I guess it comes down to, what is your vision for Listmonk?   Everyone wants a free/cheap mailchimp alternative.   
If you added an smtp/mta, you'd get a *lot* more users potentially.     Do you and/or this community want to deal with all the support/education that will be required by adding that feature? as @andreafa mentioned.
The reality is that email and deliverability are hard, and ESPs do a good job of making the difficult bits manageable, even if they are often overpriced.   It is just so easy and cheap to setup SES or sendgrid and focus on what you are good at/want to be doing.

**Comment by atomtigerzoo** [comment]
I would leave it out. There are loads of offers one can choose to send out the actual mails. It will be taken care of reliability, reputation, spam, bounces etc. I would never use my own SMPT or IP to send out lots of mails because it could harm the reputation of the whole server/network/provider.

As one said, it will also raise new issues on deliver/reputation/block/sending problems.

**Comment by knadh** [comment]
Thank you everyone for the feedback. Until there is a strong enough reason to revisit this, putting this idea on the back burner.

**Comment by Martech-WorkShop** [comment]
Sorry about the issue scavenging, but I couldn't agree more with the voices against integrating an MTA as part of ListMonk.

Besides what has already been said about tons of extra coding and legit support work, becoming responsible for a new wave of unsolvable issues related to email deliverability, etc...

I wanted to include the issue of enabling spammers. Spam is a real problem, it affects everyone, yes, but marketers, content producers and other legitimate mass emailers are the most affected, that's you and me.

Keeping a production MTA running is hard, commercial MTAs out there, have plenty of expertise, not only properly sending emails, but also preventing bad actors from using their services and damaging their reputation.

By NOT adding an MTA to ListMonk, you force (almost) everyone to go through one of the existing MTAs and hence through those spam prevention filters, which is good for everyone.
With current prices for a quality MTA as low as $1 per 1k emails sent (AWS-SES), there's no reason to facilitate a more cost-effective way anymore.
